src/Plugins/ComplexCore/src/ComplexCore/Filters/Algorithms/ImportBinaryCTNorthstar.cpp

```diff
@@ -0,0 +1,148 @@
+#include "ImportBinaryCTNorthstar.hpp"
+
+#include "complex/Common/ScopeGuard.hpp"
+#include "complex/DataStructure/DataArray.hpp"
+#include "complex/DataStructure/Geometry/ImageGeom.hpp"
+
+using namespace complex;
+
+namespace
+{
+#if defined(_MSC_VER)
+#define FSEEK64 _fseeki64
+#else
+#define FSEEK64 std::fseek
+#endif
+
+// -----------------------------------------------------------------------------
+Result<> SanityCheckFileSizeVersusAllocatedSize(size_t allocatedBytes, size_t fileSize)
+{
+  if(fileSize < allocatedBytes)
+  {
+    return MakeErrorResult(-4000, fmt::format("File size ({} bytes) is less than allocated size ({} bytes).", fileSize, allocatedBytes));
+  }
+
+  // File Size and Allocated Size are equal, so we  are good to go
+  return {};
+}
+
+// -----------------------------------------------------------------------------
+Result<> ReadBinaryCTFiles(DataStructure& dataStructure, const IFilter::MessageHandler& messageHandler, const std::atomic_bool& shouldCancel, const ImportBinaryCTNorthstarInputValues* inputValues)
+{
+  auto& geom = dataStructure.getDataRefAs<ImageGeom>(inputValues->ImageGeometryPath);
+  geom.setUnits(static_cast<IGeometry::LengthUnit>(inputValues->LengthUnit));
+
+  auto& density = dataStructure.getDataRefAs<Float32Array>(inputValues->DensityArrayPath);
+  density.fill(0xABCDEF);
+
+  usize deltaX = inputValues->EndVoxelCoord[0] - inputValues->StartVoxelCoord[0] + 1;
+
+  usize zShift = 0;
+  int32 fileIndex = 1;
+
+  for(const auto& dataFileInput : inputValues->DataFilePaths)
+  {
+    fs::path dataFilePath = inputValues->InputHeaderFile.parent_path() / dataFileInput.first;
+    const usize fileSize = fs::file_size(dataFilePath);
+    // allocated bytes should be the x * y dims * number of slices in the current data file....not necessarily the size of the whole density array
+    usize allocatedBytes = inputValues->OriginalGeometryDims[0] * inputValues->OriginalGeometryDims[1] * dataFileInput.second * sizeof(float32);
+
+    Result<> result = SanityCheckFileSizeVersusAllocatedSize(allocatedBytes, fileSize);
+    if(result.invalid())
+    {
+      return MakeErrorResult(-38705, fmt::format("The size of file '{}' on the file system ({} bytes) is less than the stated size in the binary CT header. ({} bytes).", dataFilePath.string(),
+                                                 fileSize, allocatedBytes));
+    }
+
+    FILE* f = fopen(dataFilePath.string().c_str(), "rb");
```

Comment:
OPTIONAL: A lot of the buffering would be handled and could maybe be faster if you use fstream, that being said this code seems to work alright and be buffered appropriately so this is definitely an optional change.